### DISCUSSION TITLE
Quarkus Dev services on openshift or kubernetes

### DISCUSSION THREAD

**Discussion by fjlobaton** [quarkusio/quarkus#30907] [Q&A]
Hi, I'm currently developing and app on quarkus, and now i need a way to setup the app in a remote server so the front end developer i'm working with can access the api and make tests without running quarkus in his computer.

Currently i'm using the dev services to start postgres and keycloak with docker in my pc. there is something similar that can be done with the openshift or kubernetes plugins for quarkus? 
i searched the docs and it only talks about deployment in like a final release, not while doing tests and the like. and if there is no automatic way to fire up both services, has someone done something similar that could steer me in the rigth direcction

Thanks

[a comment by an automated account is not shown]

**Comment by iocanel** [comment]
This should work already for kubernetes when you add the `quarkus-kubernetes-client` extension to your project.

**Reply by maxandersen** [reply]
@iocanel I think question is how devservices can be started when running app in kubernetes/openshift.

For that there are following options: 

1) allow docker-in-docker setup on kubernetes. This is possible but you might not be allowed if you do not have cluster admin rights in your openshift/kubernetes cluster.

2) enable https://github.com/joyrex2001/kubedock to treat docker commands to be launching kubernetes containers.

3) use a service like TestContainers Cloud that let you have docker commands to then be run in a remote location outside your cluster.

Unfortunately we are kind missing some documentation on these options; but wanted to mention them all as they all are doable. if op or someone does make it work would be be awesome if wanted to make PR for updates to the guides :)

**Reply by iocanel** [reply]
I think you are right! :blush:

**Reply by fjlobaton** [reply]
Thank you for the information. number 2 seems the easier aproach, but cant seem to find somewhere where they explain how to set up kubedock, if you have any knowledge of this please let me know. i will also try to make the openshift deployment as it will eventually will go to production migth as well and start working on that
